Components/Modal/Modal.tsx

```diff
@@ -0,0 +1,195 @@
+import * as React from 'react';
+import styled from 'styled-components';
+import ReactDOM from 'react-dom';
+
+export interface ModalProps {
+    handleClose: () => void;
+    show: boolean;
+    modalContent: any;
+    style?: object;
+    title?: string;
+    titleClass?: string;
+    titleStyle?: object;
+}
+
+//styles container of modal (background).
+var ModalWrapper = styled.div`
+    width: 100vw;
+    height: 100vh;
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    background-color: rgba(0, 0, 0, 0.5);
+    z-index: 10000;
+    position: fixed;
+    top: 0;
+    overflow: hidden;
+`;
+
+//styles content section of modal (foreground).
+var ContentContainer = styled.div`
+    max-width: 768px;
+    max-height: 95vh;
+    overflow: auto;
+    background-color: #fff;
+    z-index: 10001;
+    border-radius: .2rem;
+    padding: 1rem;
+    `;
+
+//Wraps the header of the content section.  Contains StyledTitle and CloseIcon.    
+var HeaderWrapper = styled.div`
+    display: flex;
+    margin-bottom: 1rem;
+    justify-content: space-between;
+`;
+
+var StyledTitle = styled.label`
+    font-size: 24px;
+    font-weight: bold;
+`;
+
+//X icon that calls handleClose from props
+var CloseIcon = styled.span`
+    background: url(data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAlgAAAJYCAYAAAC+ZpjcAAAABmJLR0QA/wD/AP+gvaeTAAAQ+0lEQVR4nO3Y3W0
+dRxIG0PYmsMQGYAjOwSk4WWbgCBYWNgQ/OANnsPtAcUWJV5czd/qnqvocoKHXi+mq/j6xNQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
```

Comment:
Could we do this in the CSS instead with `:before` and `:after` tags?  I'd rather not have huge base64 encoded image chilling in our library.

Reply:
👍  `:before` & `:after` would be good here (animatable too)

Reply:
Although we should prob. only provide the function and allow them to attach it to their onClick to their own UI element.